compiler/rustc_ast/src/token.rs

```diff
@@ -59,7 +135,33 @@ pub enum Delimiter {
     /// "macro variable" `$var`. It is important to preserve operator priorities in cases like
     /// `$var * 3` where `$var` is `1 + 2`.
     /// Invisible delimiters might not survive roundtrip of a token stream through a string.
-    Invisible,
+    Invisible(InvisibleOrigin),
+}
+
+impl Delimiter {
+    // Should the parser skip these delimiters? Only happens for certain kinds
+    // of invisible delimiters. Ideally this function will eventually disappear
+    // and no invisible delimiters will be skipped.
+    #[inline]
+    pub fn skip(&self) -> bool {
+        use InvisibleOrigin::*;
```

Comment:
Could you avoid glob importing enums?

I've seen you doing it as a part of cleanups to various compiler crates, but I don't think it's an improvement (only in exceptional cases maybe, but nothing in this PR looks like that).

Reply:
Ok, I'll remove this one.